terminus-ui/src/date-range/date-range.component.spec.ts

```diff
@@ -14,309 +20,231 @@ import { TsDateRangeComponent } from './date-range.component';
+      const fixture = createComponent(testComponents.Params);
+      const hostInstance = fixture.componentInstance;
+      fixture.detectChanges();
+      const startInputInstance = fixture.debugElement.queryAll(By.css('ts-input'))[0].componentInstance;
+      const endInputInstance = fixture.debugElement.queryAll(By.css('ts-input'))[1].componentInstance;
 
-  });
+      expect(endInputInstance.maxDate).toEqual(hostInstance.endMax);
+      expect(endInputInstance.minDate).toEqual(hostInstance.endMin);
 
+      expect(startInputInstance.isDisabled).toEqual(true);
+      expect(endInputInstance.isDisabled).toEqual(true);
 
-  describe(`startDateControl`, () => {
-
-    test(`should use fallback control if one was not passed in`, () => {
-      expect(component.startDateControl instanceof AbstractControl).toBeTruthy();
-    });
-
-  });
+      expect(startInputInstance.startingView).toEqual(hostInstance.startingView);
+      expect(endInputInstance.startingView).toEqual(hostInstance.startingView);
 
+      expect(startInputInstance.maxDate).toEqual(hostInstance.startMax);
+      expect(startInputInstance.minDate).toEqual(hostInstance.startMin);
 
-  describe(`endDateControl`, () => {
+      expect(startInputInstance.theme).toEqual(hostInstance.theme);
+      expect(endInputInstance.theme).toEqual(hostInstance.theme);
 
-    test(`should use fallback control if one was not passed in`, () => {
-      component.theme = 'accent';
-      expect(component.endDateControl instanceof AbstractControl).toBeTruthy();
-      expect(component.theme).toEqual('accent');
+      expect.assertions(10);
     });
 
   });
 
 
-  describe(`blur events`, () => {
-    const date = new Date(2018, 1, 1);
-
-    test(`should set the start date`, () => {
-      component.startBlur(undefined);
-      expect(component.endMinDate$.getValue()).toEqual(undefined);
-
-      component.startBlur(date);
-      expect(component.endMinDate$.getValue()).toEqual(date);
-    });
-
-
-    test(`should set the end date`, () => {
-      component.endBlur(undefined);
-      expect(component.startMaxDate$.getValue()).toEqual(undefined);
-
-      component.endBlur(date);
-      expect(component.startMaxDate$.getValue()).toEqual(date);
-    });
+  test(`should work without a form group`, function() {
+    const fixture = createComponent(testComponents.NoFormGroup);
+    fixture.detectChanges();
+    const startInputInstance = fixture.debugElement.queryAll(By.css('ts-input'))[0].componentInstance;
+    typeInElement('3-4-2019', startInputInstance.inputElement.nativeElement);
+    fixture.detectChanges();
 
+    expect(fixture.componentInstance.startSelected).toHaveBeenCalled();
   });
 
+});
 
-  describe(`isDisabled`, () => {
-
-    test(`should set the disabled flag`, () => {
-      expect(component.isDisabled).toEqual(false);
 
-      component.isDisabled = true;
 
-      expect(component.isDisabled).toEqual(true);
-    });
 
-  });
+/**
```

Comment:
There are four new lines above this one

Reply:
I often use 4 line breaks between large sections (ie between imports and code, or tests and helpers etc). Our linter will allow up to 4, then trigger a warning.